### PR TITLE
Add ZoomableImageCitationWrapper component + bump headless ui version

### LINKED ISSUE #5854: Create component to zoom citation

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This fixes https://github.com/dust-tt/dust/issues/5854.

This PR adds a `ZoomableImageCitationWrapper` component to wrap a `Citation` component and add a zoom effect when clicked. This is required as part of the vision initiative.

It looks like this:

zoomableImage

I bumped headless UI to the latest version and made sure that it was not breaking anything on our side.

### CODE REVIEW

File: sparkle/package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
   "dependencies": {
     "@emoji-mart/data": "^1.1.2",
     "@emoji-mart/react": "^1.1.1",
-    "@headlessui/react": "^1.7.17",
```
Comment (JulesBelveze) [inline sparkle/package.json:82]: Heads up: are we okay with two different versions of headless in `sparkle` and `front`?

**Review by JulesBelveze — APPROVED** [review]
Very cool 🙏🏼

File: sparkle/package.json (hunk 1 above)
Comment (flvndvd) [reply sparkle/package.json:82]: Ahah I guess we are not, I'll bump the front's version as well but it requires more work. FWIW it's okay-ish as it does not break any existing behavior.

Comment (JulesBelveze) [reply sparkle/package.json:82]: Can help with that if needed 👍🏼

Comment (flvndvd) [reply sparkle/package.json:82]: All yours!

**Pull request merged**